src/test/java/org/opentripplanner/middleware/controllers/api/ManageLegTraversalTest.java

```diff
@@ -148,202 +148,150 @@ private static Stream<Arguments> createTrace() {
+        final int NORTH_BEARING = 0;
+        final int NORTH_EAST_BEARING = 45;
+        final int SOUTH_BEARING = 180;
+        final int SOUTH_WEST_BEARING = 225;
+        final int NORTH_WEST_BEARING = 315;
+        Leg adairAvenueToMonroeDriveLeg = adairAvenueToMonroeDriveItinerary.legs.get(0);
+        List<Step> walkSteps = adairAvenueToMonroeDriveLeg.steps;
+        Coordinates originCoords = new Coordinates(adairAvenueToMonroeDriveLeg.from);
+        Coordinates destinationCoords = new Coordinates(adairAvenueToMonroeDriveLeg.to);
+        String destinationName = adairAvenueToMonroeDriveLeg.to.name;
         Step stepOne = walkSteps.get(0);
         Coordinates stepOneCoords = new Coordinates(stepOne);
         Step stepTwo = walkSteps.get(1);
         Coordinates stepTwoCoords = new Coordinates(stepTwo);
         Step stepThree = walkSteps.get(2);
         Coordinates stepThreeCoords = new Coordinates(stepThree);
-        Step stepFour = walkSteps.get(3);
-        Coordinates stepFourCoords = new Coordinates(stepFour);
-        Step stepFive = walkSteps.get(4);
-        Coordinates stepFiveCoords = new Coordinates(stepFive);
-
-        Leg rockSpringsLeg = edmundParkDriveToRockSpringsItinerary.legs.get(0);
-        List<Step> rockSpringsSteps = rockSpringsLeg.steps;
-
-        Coordinates fourthGeoPoint = new Coordinates(33.79352,-84.34148);
-        Coordinates sixthGeoPoint = new Coordinates(33.79332,-84.34099);
-        Step rockSpringsStepTwo = rockSpringsSteps.get(1);
-
-        Leg adairAvenueToMonroeDriveLeg = adairAvenueToMonroeDriveItinerary.legs.get(0);
-        List<Step> adairAvenueToMonroeDriveLegSteps = adairAvenueToMonroeDriveLeg.steps;
-        Step virginiaAvenue = adairAvenueToMonroeDriveLegSteps.get(5);
 
+        Step virginiaAvenue = walkSteps.get(5);
         Coordinates pointBeforeTurn = new Coordinates(33.78151,-84.36481);
         Coordinates virginiaAvenuePoint = new Coordinates(virginiaAvenue);
 
         return Stream.of(
             Arguments.of(
                 new TurnTrace(
-                    adairAvenueToMonroeDriveItinerary,
                     createPoint(pointBeforeTurn, 8, calculateBearing(pointBeforeTurn, virginiaAvenuePoint)),
                     new TripInstruction(10, virginiaAvenue).build(),
                     false,
-                    "Approaching left turn on Virginia Avenue."
+                    "Approaching left turn on Virginia Avenue (Test to make sure turn is not missed)."
                 )
             ),
             Arguments.of(
                 new TurnTrace(
-                    adairAvenueToMonroeDriveItinerary,
                     createPoint(pointBeforeTurn, 16, calculateBearing(pointBeforeTurn, virginiaAvenuePoint)),
                     new TripInstruction(2, virginiaAvenue).build(),
                     false,
-                    "Turn left on to Virginia Avenue."
+                    "Turn left on to Virginia Avenue (Test to make sure turn is not missed)."
```

Comment:
We are missing a case where the position is immediately after a turn, but still within the "upcoming" radius. In that case, the instruction should be on-track. On the field, the upcoming turn instruction still appears after having taking the turn as instructed.

Reply:
Add this case:
```java
            Arguments.of(
                new TurnTrace(
                    createPoint(pointAfterTurn, 0, calculateBearing(pointAfterTurn, virginiaAvenuePoint)),
                    NO_INSTRUCTION,
                    false,
                    "After turn left on to Virginia Avenue should not produce turn instruction."
                )
            ),
```
where 
```java
        Coordinates pointAfterTurn = new Coordinates(33.78165, -84.36484);
```